### PR TITLE
[Feature] Support Tencent cos filesystem

### PR DESCRIPTION
**Pull request by letian-jiang**
```sql
CREATE EXTERNAL CATALOG `ice_hadoop_cos`
PROPERTIES (
	"type"  =  "iceberg",
	"iceberg.catalog.type"  =  "hadoop",
	"iceberg.catalog.warehouse" = "cosn://xxx/iceberg_catalog/",
	"tencent.cos.access_key"  =  "access_key",
	"tencent.cos.secret_key"  =  "secret_key",
	"tencent.cos.endpoint"  =  "cos.ap-shanghai.myqcloud.com"
)
```

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/credential/CloudType.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 18 / 49 (36.73%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/credential/tencent/TencentCloudConfiguration.java|0|17|00.00%|[31, 32, 33, 34, 39, 40, 41, 42, 43, 44, 48, 49, 50, 54, 59, 64, 65]|
|:large_blue_circle:|com/starrocks/credential/tencent/TencentCloudCredential.java|9|22|40.91%|[42, 43, 44, 45, 46, 47, 48, 58, 59, 60, 61, 65, 74]|
|:large_blue_circle:|com/starrocks/credential/tencent/TencentCloudConfigurationProvider.java|8|9|88.89%|[40]|
|:large_blue_circle:|com/starrocks/credential/CloudType.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/credential/tencent/TencentCloudCredential.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        checkNotNull(secretKey);
+        checkNotNull(endpoint);
+        this.accessKey = accessKey;
+        this.secretKey = secretKey;
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    public void applyToConfiguration(Configuration configuration) {
+        configuration.set("fs.cosn.impl", "org.apache.hadoop.fs.CosFileSystem");
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/credential/tencent/TencentCloudCredential.java:42]: specific reason to use CosNFileSystem to access cos? will aws sdk work or not?

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/credential/tencent/TencentCloudCredential.java:42]: it test worked. FE using CosNFileSystem to access cos/ BE using was SDK to access cos

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/credential/tencent/TencentCloudCredential.java:42]: FE contains aws java sdk, if it works, maybe we can use aws java sdk to the same work, unless there are some advantages that CosFileSystem provides? b.t.w., this configuration will not work unless the jar implemented CosFileSystem is packed into FE as well which makes FE fatter and fatter.

Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/credential/tencent/TencentCloudCredential.java:42]: I will update related dependencies later.

> FE contains aws java sdk, if it works, maybe we can use aws java sdk to the same work, unless there are some advantages that CosFileSystem provides? b.t.w., this configuration will not work unless the jar implemented CosFileSystem is packed into FE as well which makes FE fatter and fatter.

Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/credential/tencent/TencentCloudCredential.java:42]: If [this](https://cloud.tencent.com/document/product/436/56971) is enabled, using s3 sdk in FE would not work appropriately.

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 18 / 49 (36.73%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/credential/tencent/TencentCloudConfiguration.java|0|17|00.00%|[31, 32, 33, 34, 39, 40, 41, 42, 43, 44, 48, 49, 50, 54, 59, 64, 65]|
|:large_blue_circle:|com/starrocks/credential/tencent/TencentCloudCredential.java|9|22|40.91%|[42, 43, 44, 45, 46, 47, 48, 58, 59, 60, 61, 65, 74]|
|:large_blue_circle:|com/starrocks/credential/tencent/TencentCloudConfigurationProvider.java|8|9|88.89%|[40]|
|:large_blue_circle:|com/starrocks/credential/CloudType.java|1|1|100.00%|[]|

**Comment by letian-jiang** [comment]
PTAL @kevincai @Smith-Cruise

**Review by kevincai — APPROVED** [review]

**Review by Smith-Cruise — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 49 / 49 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/credential/CloudType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/credential/tencent/TencentCloudConfigurationProvider.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/credential/tencent/TencentCloudCredential.java|22|22|100.00%|[]|
|:large_blue_circle:|com/starrocks/credential/tencent/TencentCloudConfiguration.java|17|17|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Youngwb — APPROVED** [review]

File: fe/fe-core/pom.xml [hunk 2]
```diff
@@ -605,6 +605,18 @@ under the License.
             <artifactId>hadoop-aws</artifactId>
         </dependency>
 
+        <!-- https://mvnrepository.com/artifact/com.qcloud.cos/hadoop-cos -->
```
Comment (dirtysalt) [inline fe/fe-core/pom.xml:608]: why it's also used in fe/pom.xml?

**Review by dirtysalt — APPROVED** [review]

**Comment by letian-jiang** [comment]
@mergify backport branch-3.2

**Pull request merged**